### PR TITLE
Network packet timing statistics

### PR DESCRIPTION
**Pull request by jukkar**
This adds infrastructure to measure how long it takes for a network packet to pass the network stack. It is also possible to collect TX/RX statistics from various key points in the network stack.

### CODE REVIEW

**Review by rlubos — APPROVED** [review]

**Review by tbursztyka — APPROVED** [review]

File: include/net/net_stats.h [hunk 1]
```diff
@@ -323,6 +327,10 @@ struct net_stats {
 	/** Network packet TX time detail statistics */
 	struct net_stats_tx_time tx_time_detail[NET_PKT_DETAIL_STATS_COUNT];
 #endif
+#if defined(CONFIG_NET_PKT_RXTIME_STATS_DETAIL)
+	/** Network packet RX time detail statistics */
+	struct net_stats_tx_time rx_time_detail[NET_PKT_DETAIL_STATS_COUNT];
```
Comment (mnkp) [inline include/net/net_stats.h:332]: This should likely be `struct net_stats_rx_time`.

Also it looks like this field will be only visible if `CONFIG_NET_PKT_TXTIME_STATS` is selected as well due to
```
#if defined(CONFIG_NET_CONTEXT_TIMESTAMP) || \
					defined(CONFIG_NET_PKT_TXTIME_STATS)
```
a few lines above.

File: subsys/net/lib/sockets/sockets.c [hunk 2]
```diff
@@ -851,6 +853,36 @@ static int sock_get_pkt_src_addr(struct net_pkt *pkt,
...
+void net_socket_update_tc_rx_time(struct net_pkt *pkt, uint32_t end_tick)
+{
+	net_pkt_set_rx_stats_tick(pkt, end_tick);
+
+	net_stats_update_tc_rx_time(net_pkt_iface(pkt),
+				    net_pkt_priority(pkt),
+				    net_pkt_timestamp(pkt)->nanosecond,
+				    end_tick);
+
+	if (IS_ENABLED(CONFIG_NET_PKT_TXTIME_STATS_DETAIL)) {
```
Comment (mnkp) [inline subsys/net/lib/sockets/sockets.c:865]: Should be `CONFIG_NET_PKT_RXTIME_STATS_DETAIL`

Comment (jukkar) [reply subsys/net/lib/sockets/sockets.c:865]: > Should be `CONFIG_NET_PKT_RXTIME_STATS_DETAIL`

Indeed, good catch

File: include/net/net_stats.h (hunk 1 above)
Comment (jukkar) [reply include/net/net_stats.h:332]: Good points, I will send a fix for these.

**Pull request merged**